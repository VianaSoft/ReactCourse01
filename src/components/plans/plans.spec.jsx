/* eslint-disable no-undef */
import { render, screen } from '@testing-library/react';

import { Plans } from '.';

const props = {
  plans: [
    {
      Id: '7c6c51ad-53bb-4e0b-8100-51e2920ce4e8',
      Name: 'Free',
      DisplayHome: false,
      MaxHome: 0,
      MaxAdvert: 1,
      MaxPhoto: 6,
      MaxVideo: 0,
      Priority: 18,
      TimeRange: 7,
      Price: 0.0,
      Total: 0.0,
      PlanType: {
        Id: '60c10154-73a0-4617-a260-a326e6b653c8',
        Name: 'Individual',
        IsEnable: true,
      },
      TimeType: {
        Id: '2f9f3ed6-2313-4d98-8a24-4c0468946c07',
        Name: 'Dia(s)',
        IsEnable: true,
      },
      IsEnable: true,
    },
    {
      Id: '05b4eacf-ee48-476a-9ad2-54c98ba66615',
      Name: 'Cl√°ssico',
      DisplayHome: false,
      MaxHome: 0,
      MaxAdvert: 1,
      MaxPhoto: 6,
      MaxVideo: 0,
      Priority: 17,
      TimeRange: 1,
      Price: 4.0,
      Total: 4.0,
      PlanType: {
        Id: '60c10154-73a0-4617-a260-a326e6b653c8',
        Name: 'Individual',
        IsEnable: true,
      },
      TimeType: {
        Id: '9dad9819-e846-44fb-b427-9721265f184e',
        Name: 'Mese(s)',
        IsEnable: true,
      },
      IsEnable: true,
    },
  ],
};

describe('<Plans />', () => {
  it('shoud render plans items', () => {
    render(<Plans {...props} />);

    expect(screen.getAllByText(/Individual/i)).toHaveLength(props.plans.length);
  });

  it('Should not render plans', () => {
    render(<Plans />);

    expect(screen.queryAllByAltText(/Individual/i)).toHaveLength(0);
    expect(screen.queryAllByAltText(/Corporativo/i)).toHaveLength(0);
    expect(screen.queryByText(/Individual/i)).not.toBeInTheDocument();
    expect(screen.queryByText(/Corporativo/i)).not.toBeInTheDocument();
  });

  it('Should match snapshot', () => {
    const { container } = render(<Plans {...props} />);

    expect(container).toMatchSnapshot();
  });
});
