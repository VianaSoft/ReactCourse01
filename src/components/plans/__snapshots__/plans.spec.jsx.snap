// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Plans /> Should match snapshot 1`] = `
<div>
  <div
    class="plans"
  >
    <div
      class="plan"
    >
      <div
        class="plan-title"
      >
        <strong>
          Free
        </strong>
      </div>
      <div
        class="plan-content"
      >
        <div
          class="plan-content-e"
        >
          <p>
            Tipo do plano
          </p>
          <p>
            Aparece na Home
          </p>
          <p>
            Anúncios
          </p>
          <p>
            Fotos
          </p>
          <p>
            Vídeos
          </p>
          <p>
            Prioridade
          </p>
          <p>
            Duração 
            Dia(s)
          </p>
          <p>
            Preço (mês)
          </p>
        </div>
        <div
          class="plan-content-d"
        >
          <p>
            <strong>
              Individual
            </strong>
          </p>
          <p>
            <strong>
              Não
            </strong>
          </p>
          <p>
            <strong>
              1
            </strong>
          </p>
          <p>
            <strong>
              6
            </strong>
          </p>
          <p>
            <strong>
              0
            </strong>
          </p>
          <p>
            <strong>
              18
            </strong>
          </p>
          <p>
            <strong>
              7
            </strong>
          </p>
          <p>
            <strong>
              0,00 €
              +IVA
            </strong>
          </p>
        </div>
      </div>
    </div>
    <div
      class="plan"
    >
      <div
        class="plan-title"
      >
        <strong>
          Clássico
        </strong>
      </div>
      <div
        class="plan-content"
      >
        <div
          class="plan-content-e"
        >
          <p>
            Tipo do plano
          </p>
          <p>
            Aparece na Home
          </p>
          <p>
            Anúncios
          </p>
          <p>
            Fotos
          </p>
          <p>
            Vídeos
          </p>
          <p>
            Prioridade
          </p>
          <p>
            Duração 
            Mese(s)
          </p>
          <p>
            Preço (mês)
          </p>
        </div>
        <div
          class="plan-content-d"
        >
          <p>
            <strong>
              Individual
            </strong>
          </p>
          <p>
            <strong>
              Não
            </strong>
          </p>
          <p>
            <strong>
              1
            </strong>
          </p>
          <p>
            <strong>
              6
            </strong>
          </p>
          <p>
            <strong>
              0
            </strong>
          </p>
          <p>
            <strong>
              17
            </strong>
          </p>
          <p>
            <strong>
              1
            </strong>
          </p>
          <p>
            <strong>
              4,00 €
              +IVA
            </strong>
          </p>
        </div>
      </div>
    </div>
  </div>
</div>
`;
